xmpp-vala/src/module/tls.vala

```diff
@@ -25,10 +25,15 @@ namespace Xmpp.Tls {
                     var io_stream = stream.get_stream();
                     if (io_stream == null) return;
                     var conn = TlsClientConnection.new(io_stream, identity);
+                    var pinned_certificate = stream.pinned_certificate;
+                    var flag = stream.get_flag(Flag.IDENTITY);
+                    conn.accept_certificate.connect((conn, server_cert, errors) => {
+                        flag.used_pinned_certificate = true;
```

Comment:
This is confusing. This flag basically says that initial verification failed, not that a pinned certificate was used.

Reply:
@ThibG are you aware of any method to extract the public key from the certificate in Vala?

Reply:
@rigelk sadly not, neither glib nor gcr seem to expose anything useful for that. I fear doing that would require exporting the public cert and writing your own code to manually extract it :/

Reply:
@ThibG that's what I saw before asking.
If I find some time to examine this problem, should I use [openssl](https://raw.githubusercontent.com/nemequ/vala-extra-vapis/master/openssl.vapi) bindings or [gcrypt](https://raw.githubusercontent.com/nemequ/vala-extra-vapis/master/gcrypt.vapi) bindings? Both seem to lack any function related to certificate management, as of now anyway. 

I could add the required bindings though.